mu4e/mu4e-icalendar.el

```diff
@@ -0,0 +1,150 @@
+	      (replace-match "\\1\n \\2")
+	      (goto-char (line-beginning-position)))))
+
+        (with-current-buffer (get-buffer-create gnus-icalendar-reply-bufname)
+          (delete-region (point-min) (point-max))
+          (insert reply)
+          (fold-icalendar-buffer)
+          (when (string= (downcase charset) "utf-8")
+            (decode-coding-region (point-min) (point-max) 'utf-8))
+          (mu4e-icalendar-reply-ical msg event status (buffer-name)))
+
+        ;; Back in article buffer
+        (setq-local gnus-icalendar-reply-status status)
+        (when gnus-icalendar-org-enabled-p
+          (gnus-icalendar--update-org-event event status))
+        (when mu4e-icalendar-diary-file
+          (mu4e~icalendar-insert-diary event status
+                                       mu4e-icalendar-diary-file))))))
+
+(defun mu4e~icalendar-delete-citation ()
+  (delete-region (point-min) (point-max)))
+
+(defun mu4e~icalendar-trash-message (msg)
+  "Trash the message MSG."
+  (with-current-buffer mu4e~headers-buffer-name
+    (let* ((docid (mu4e-message-field msg :docid))
+           (markdescr (assq 'trash mu4e-marks))
+           (action (plist-get (cdr markdescr) :action))
+           (target (mu4e-get-trash-folder msg)))
+      (run-hook-with-args 'mu4e-mark-execute-pre-hook 'trash msg)
+      (funcall action docid msg target))))
+
+(defun mu4e-icalendar-reply-ical (original-msg event status buffer-name)
+  (let ((message-signature nil))
+    (let ((mu4e-compose-cite-function #'mu4e~icalendar-delete-citation)
+          (mu4e-sent-messages-behavior 'delete)
+          (mu4e-compose-reply-recipients 'sender))
+      (mu4e~compose-handler 'reply original-msg))
+    ;; Make sure the recipient is the organizer
+    (let ((organizer (gnus-icalendar-event:organizer event)))
+      (unless (string= organizer "")
+        (message-goto-to)
+        (delete-region (line-beginning-position) (line-end-position))
+        (insert "To: " organizer)))
+    (message-goto-body)
+    (mml-insert-multipart "alternative")
+    (mml-insert-part "text/plain")
+    (let ((reply-event (gnus-icalendar-event-from-buffer
+                        buffer-name mu4e-user-mail-address-list)))
+      (insert (gnus-icalendar-event->gnus-calendar reply-event status)))
+    (forward-line 1); move past closing tag
+    (mml-attach-buffer buffer-name "text/calendar; method=REPLY; charset=utf-8")
+    (message-goto-subject)
+    (delete-region (line-beginning-position) (line-end-position))
+    (insert "Subject: " (capitalize (symbol-name status))
+            ": " (gnus-icalendar-event:summary event))
+    (set-buffer-modified-p nil); not yet modified by user
+    (when mu4e-icalendar-trash-after-reply
+      ;; `mu4e~switch-back-to-mu4e-buffer' was executed.
+      (push
+       (lexical-let ((msg original-msg))
+         #'(lambda ()
+             (when (and (not (eq mu4e-split-view 'single-window))
+                        (buffer-live-p (mu4e-get-view-buffer)))
```

Comment:
Here some help would be welcome because `(eq mu4e-split-view 'single-window)` appears at quite a few places.  What we are trying to do is, after the reply is sent, to go back to the original message in order to move to the next (if possible) and then delete it.

Reply:
In order to avoid this test, it would be good to run after `mu4e~switch-back-to-mu4e-buffer` in `mu4e-sent-handler`.  This latter function is applied at the very end of the process, so a hook (e.g. in `message-send-actions`) will run before it.  It is neither possible to have a local hook in the compose buffer since `mu4e-sent-handler` will be executed after this buffer has been closed.  I could add a hook to `message-sent-hook` to override `mu4e-sent-func` with a function of my choice but the problem is that I need the original message and thus must partially apply the function.  This works and I can do that if it is fine with you.  I've pushed this idea in the last commit so you can see what it looks like.

Reply:
(Enabled lexical binding; see below)